guide/14-deep-learning/how_estimate_batch_size_works.ipynb

```diff
@@ -0,0 +1,251 @@
+{
```

Comment:
<p>It is possible to use the <code>estimate_batch_size</code> function to determine the <code>recommended_batchsize</code> during the data preparation stage while using the <code>prepare_data</code> function. The <code>prepare_data</code> function typically prepares the data for model training by performing tasks like data cleaning, preprocessing, and splitting the data into training, validation, and test sets. To use <code>estimate_batch_size</code> from the <code>prepare_data</code> function while creating the data object using <code>prepare_data</code>, we need to pass <code>batch_size</code> as <code>None</code>. This will call the <code>estimate_batch_size</code> function within the <code>prepare_data</code> function, passing in the relevant data and model parameters as arguments, and setting the <code>recommended_batchsize</code> as the <code>batch_size</code> for the data object. This helps ensure that the model is trained using the best possible batch size, improving its performance and reducing the likelihood of memory errors.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1533/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>